app/src/main/java/org/openobservatory/ooniprobe/test/TestAsyncTask.java

```diff
@@ -133,6 +133,10 @@ private void runTest(AbstractTest... tests) {
                 }
                 if (!interrupt) {
                     Log.d(TAG, "run next suite: " + currentSuite.getName() + " test:" + currentTest.getName());
+
+                    ThirdPartyServices.addLogExtra("name", currentTest.getName());
+                    ThirdPartyServices.addLogExtra("_result", app.getGson().toJson(result));
+
```

Comment:
I'm not sure this is the right place and thing we should be logging. What we are most interested in recording is the `input` that are being passed to the test, which it's my understanding that they are part of the `settings` object that is somehow created via the `context` and `preferencesmanager`.

I would suggest instead to log the `settings` object over here: https://github.com/ooni/probe-android/blob/555917c1deb61c6ad98168fb1a5669b54f000eff/app/src/main/java/org/openobservatory/ooniprobe/test/test/WebConnectivity.java#L24

Reply:
I thought of doing the log there but i realized that if the offender is not the `WebConnectivity` test, the only other way to capture the test name is to log the name and input here since the `WebConnectivity` and other tests will eventually get called based on their implementations of `AbstractTest.run`

Reply:
That's a good point, however we do know that the offender is `WebConnectivity` because:
1. The logs all report the WebConnectivity in the backtrace
2. The only test which takes inputs is currently WebConnectivity

Reply:
OK i get your point there. 

I went for the generic option that will both clarify the current error with `WebConnectivity` and also any future errors for any other test.

Reply:
Excellent, thanks!